dashboard/lib/pd/foorm/form_analytics_parser.rb

```diff
@@ -0,0 +1,112 @@
+module Pd::Foorm
+  # Responsible for taking Forms (ie, a survey comprised of many questions)
+  # from our in-house survey system (Foorm)
+  # and processing them into a "one row per question" format
+  # for export to CSV, and, ultimately, to our analytics database (Redshift).
+  class FormAnalyticsParser
+    include Constants
+    extend Helper
+
+    HEADERS = %w(
+      form_id
+      form_name
+      form_version
+      question_type
+      question_name
+      question_text
+      matrix_item_name
+      matrix_item_text
+      is_facilitator_specific
+      response_options
+      num_response_options
+    )
+
+    def self.reshape_all_forms_and_export_to_csv
+      CSV.open('./test_forms.csv', 'wb') do |csv|
+        csv << HEADERS
+
+        # Loads 1000 records at a time to manage loading records into memory.
+        # Probably unnecessary optimization until we have many more Forms.
+        ::Foorm::Form.find_each do |form|
+          reshaped_form = reshape_form(form)
+
+          reshaped_form.each do |question|
+            row = question.stringify_keys.values_at(*HEADERS)
+            csv << row
+          end
+        end
+      end
+    end
+
+    def self.reshape_form(form)
```

Comment:
Maybe this would make more sense as a mix-in for (instances of) the Form class?

Reply:
(I'm sorry, I never really understood how to use mixins so I don't know the answer to this question without doing more research.)